lib/ecto/adapter.ex

```diff
@@ -45,6 +45,11 @@ defmodule Ecto.Adapter do
   """
   @callback child_spec(repo, options) :: Supervisor.Spec.spec
 
+  @doc """
+  Initializes the adapter
+  """
+  @callback init(repo, options) :: {:ok, state :: term} | {:error, reason :: term}
```

Comment:
There is no need for an init callback. If the adapter needs to do extra work, then it needs to wrap the child_spec or the function in the child_spec returned by DBConnection by its function/child_spec and then execute any custom action that is necessary.